### PR TITLE
Controllers specs

### PR DESCRIPTION
**Pull request by OpondoG**
## Dear Reviewer(s)

- Thank you for taking the time to review this pull request.
- I implemented the following:

- [ ] Add RSpec as a gem to your project.
- [ ] Create a Request spec file for all your controllers.
- [ ] Make sure that for each action you check:
- [ ] If response status was correct.
- [ ] If a correct template was rendered.
- [ ] If the response body includes correct placeholder text.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -18,6 +18,8 @@ gem 'sprockets-rails'
 # Use postgresql as the database for Active Record
 gem 'pg', '~> 1.1'
 
+gem 'rails-controller-testing'
+
```
Comment (codecaiine) [inline Gemfile]: - [x] Please add RSpec gem for testing  be similar to the example below :arrow_down:  It needs to be in the :development group to expose generators and rake tasks without having to type RAILS_ENV=test. :+1: 

```
group :development, :test do
 ...
 gem 'rspec-rails'
end

```

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
```
Comment (codecaiine) [inline spec/requests/posts_spec.rb:1]: - [x] Some tests are failing . Please make sure to fix them and make all the test passed as required for this project :+1: 
Screenshot_20230208_204912

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @OpondoG , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Great testing setup :heavy_check_mark: 
- No rubocop errors :heavy_check_mark: 
- Good use of gitflow :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @OpondoG 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done on making the required changes from the previous reviewer.

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**